### PR TITLE
Add filtering on result

### LINKED ISSUE #1774: anonymous user (no logged in) can find private API using APINF search
# Reproduction Steps:
1. go to https://staging.apinf.io
2. logged in as admin
3. go to catalog page and see what APIs are private.
image
4. log out
5. click on search and look for the private API

# Result:
* link to the name of the private API appears as search result
image

# Expected result:
Even when logged in as registered users, noone can see a private API unless it is made visible to certain users by the admin, anonymous users (not logged in) should not be able to get the API with search result. Same applies for logged in user who has not permission to view the API.

### PR DESCRIPTION
**Pull request by marla-singer**
Closes #1774

### CODE REVIEW

**Comment by jykae** [comment]
Reviewing

File: search/client/search.js [hunk 1]
```diff
@@ -46,8 +46,16 @@ Template.search.created = function () {
...
 
   instance.getSearchResults = function () {
-    var query = instance.query.get();
-    return Apis.find(query).fetch();
+    // Get query result
+    const query = instance.query.get();
+    // Find api in collection
+    const searchResult = Apis.find(query).fetch();
+    // Filter that: can current user view it or not
+    const filteredResult = searchResult.filter((api) => {
```
Comment (jykae) [inline search/client/search.js]: nice solution 👍

Comment (marla-singer) [inline search/client/search.js]: @jykae  Thanks! :)

**Comment by jykae** [comment]
Line 22 search.js
nayttokuva 2016-10-13 kello 16 42 46

**Comment by marla-singer** [comment]
It makes me sad.
Did It happen then you tried to find non-existent api?

**Comment by jykae** [comment]
Check import for the above

**Comment by marla-singer** [comment]
Linting in awkward hands means terrible

**Comment by jykae** [comment]
@marla-singer I committed small fix to that awkward hands commit. onCreated seems to not work with arrow style for some reason o_O

**Comment by marla-singer** [comment]
@jykae thanks for commit. Yeah! onCreated doesn't work with arrow style %)

**Comment by jykae** [comment]
Searching on search page does not add query string to URL, but it was not part of this issue. Let's make it in another PR. I can merge this.

**Pull request merged**